File: backend/demo/src/main/java/project_z/demo/services/RoomBanService.java
package project_z.demo.services;

import java.util.List;
import java.util.UUID;

import org.springframework.stereotype.Service;

import project_z.demo.dto.RoomBanDtos.RoomBanCreateDto;
import project_z.demo.dto.RoomBanDtos.RoomBanDetailsDto;

@Service
public interface RoomBanService {
    RoomBanDetailsDto create(RoomBanCreateDto ban,Long roomId);
    List<RoomBanDetailsDto> findAllByRoom(Long roomId);
    boolean isBanned(Long roomId, UUID userId);
    void unban(UUID id);
}

File: frontend/project_z_frontend/app/features/manageRooms/api/roomBanService.ts
import { apiClient } from "~/shared/api";
import type { RoomBanCreateDto, RoomBanDetailsDto } from "../model/roomTitle.types";
import type { UserDtoWithRoomBanStatus, UserParams } from "~/entities/user/model/user.types";
import type { PageResponse } from "~/shared/types";


export interface RoomBanService {
    findAllByRoom(roomId: number): Promise<RoomBanDetailsDto[]>;
    create(roomId: number, banDto: RoomBanCreateDto): Promise<RoomBanDetailsDto>;
    isBanned(roomId: number, userId: string): Promise<boolean>;
    unban(roomId: number, roomBanId: string): Promise<void>;
    searchUsers(roomId:number, queryParameters:UserParams):Promise<PageResponse<UserDtoWithRoomBanStatus>>;
}

export const roomBanService: RoomBanService = {
    async findAllByRoom(roomId) {
        const { data } = await apiClient.get(`/rooms/${roomId}/bans`);
        return data;
    },

    async create(roomId, banDto) {
        const { data } = await apiClient.post(`/rooms/${roomId}/bans`, banDto);
        return data;
    },

    async isBanned(roomId, userId) {
        const { data } = await apiClient.get(`/rooms/${roomId}/bans/check/${userId}`);
        return data;
    },

    async unban(roomId, roomBanId) {
        await apiClient.delete(`/rooms/${roomId}/bans/${roomBanId}`);
    },
    async searchUsers(roomId,queryParameters){
        const{data}=await apiClient.get(`/rooms/${roomId}/bans/search`,
            {params:queryParameters}
        )
        return data;
    }

};
